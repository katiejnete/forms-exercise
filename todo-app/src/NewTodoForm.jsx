import React from "react";
import { useFormik } from "formik";

const NewTodoForm = ({ addTodo }) => {
  const formik = useFormik({
    initialValues: { task: "" },
    validate: (values) => {
      const errors = {};
      if (!values.task) errors.task = "Required";
      return errors;
    },
    onSubmit: (values, { resetForm }) => {
      addTodo(values);
      resetForm();
    },
  });

  return (
    <form onSubmit={formik.handleSubmit}>
      <input
        type="text"
        name="task"
        placeholder="Task Name"
        onChange={formik.handleChange}
        value={formik.values.task}
      />
      {formik.errors.task && formik.touched.task && (
        <div style={{ color: "red" }}>{formik.errors.task}</div>
      )}
      <button type="submit">Submit</button>
    </form>
  );
};

export default NewTodoForm;
