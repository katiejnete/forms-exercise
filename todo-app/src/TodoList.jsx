import React, { useState } from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";
import { v4 as uuid } from "uuid";

const TodoList = () => {
  const [todos, setTodos] = useState([]);
  const addTodo = (todo) => {
    const newTodo = { ...todo, id: uuid() };
    setTodos((todos) => [...todos, newTodo]);
  };
  const removeTodo = (id) => {
    setTodos(todos.filter((todo) => todo.id !== id));
  };
  return (
    <>
      <NewTodoForm addTodo={addTodo} />
      {todos.map((todo) => (
        <Todo
          key={todo.id}
          id={todo.id}
          removeTodo={removeTodo}
          task={todo.task}
        />
      ))}
    </>
  );
};

export default TodoList;
