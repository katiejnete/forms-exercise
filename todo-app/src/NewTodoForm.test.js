import React from "react";
import {render} from "@testing-library/react";
import NewTodoForm from "./NewTodoForm.jsx";
import {addTodo} from "./TodoList.jsx"

// smoke test
it("renders without crashing", () => {
    render(<NewTodoForm addTodo={addTodo} />);
})

// snapshot test
it("matches snapshot", () => {
    const {asFragment} = render(<NewTodoForm addTodo={addTodo} />);
    expect(asFragment()).toMatchSnapshot();
})