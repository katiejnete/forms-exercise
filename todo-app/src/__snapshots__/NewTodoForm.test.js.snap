// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <input
      name="task"
      placeholder="Task Name"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
